<launch>
  <!-- 参数声明 -->
  <arg name="exponents" default="1.0"/>
  <arg name="fxtdo_enabled" default="true"/>
  <arg name="trajectory_type" default="circle"/>

  <!-- 加载 YAML -->
  <rosparam command="load" file="$(find mmg_mppi_controller)/config/config.yaml" />

  <!-- 覆盖关键参数 -->
  <param name="exponents" value="$(arg exponents)" />
  <param name="fxtdo/enabled" value="$(arg fxtdo_enabled)" />
  <param name="simulation/traj_type" value="$(arg trajectory_type)" />

  <!-- 控制器节点 -->
  <node name="mppi_controller" pkg="mmg_mppi_controller" type="mppi_controller" output="screen" required="true" />

  <!-- TrajGenerator 节点 -->
  <node name="TrajGenerator" pkg="mmg_mppi_controller" type="traj_generator.py" output="screen" required="true" />
</launch>
